src/core/reducer.ts

```diff
@@ -121,6 +128,77 @@ function initializeDeltalog(
   };
 }
 
+/**
+ * ExtractTransientsFromState
+ *
+ * Split out transients from the a TransientState
+ */
+function ExtractTransients(
+  transientState: TransientState
```

Comment:
We should probably mark that this can be `null` in the signature:

```suggestion
  transientState: TransientState | null
```

(Our tsconfig doesn’t have strict null checking, so `null` and `undefined` are just ignored as possibilities by the compiler, but I try to be specific in newer code in the hope of one day being able to enable stricter typechecking.)